docs/setup.asciidoc

```diff
@@ -34,19 +41,104 @@ https://www.nuget.org/packages/Elastic.Apm.SqlClient[**Elastic.Apm.SqlClient**]:
 
 This package contains https://www.nuget.org/packages/System.Data.SqlClient[System.Data.SqlClient] and https://www.nuget.org/packages/Microsoft.Data.SqlClient[Microsoft.Data.SqlClient] monitoring related code.
 
+
 [float]
 === Quick start
 
+* <<setup-dotnet-net-core>>
 * <<setup-asp-net-core>>
 * <<setup-asp-net>>
 * <<setup-ef6>>
 * <<setup-sqlclient>>
 * <<setup-general>>
 
+
+[float]
+[[setup-dotnet-net-core]]
+==== NET Core
+
+On .NET Core the agent can be registered on the `IHostBuilder`. This applies to both ASP.NET Core and to other .NET Core applications that depend on `IHostBuilder` like https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/hosted-services[background tasks]. In this case you need to reference the `Elastic.Apm.NetCoreAll` package.
+
+
+[source,csharp]
+----
+using Elastic.Apm.NetCoreAll;
+
+namespace MyApplication
+{
+  public class Program
+  {
+    public static IHostBuilder CreateHostBuilder(string[] args) =>
+        Host.CreateDefaultBuilder(args)
+            .ConfigureWebHostDefaults(webBuilder => { webBuilder.UseStartup<Startup>(); })
+            .UseAllElasticApm();
+
+    public static void Main(string[] args) => CreateHostBuilder(args).Build().Run();
+  }
+}
+----
+
+With the `UseAllElasticApm()` the agent with all its components is turned on. On ASP.NET Core it'll automatically capture incoming requests, database calls through supported technologies, outgoing HTTP requests and so on.
+
+Additionally the `UseAllElasticApm` will also add an `ITracer` to the Dependency Injection system. This means you can use the <<public-api>> in your code like this:
```

Comment:
We are in a `Quick start` section here, and honestly I'm not sure where to stop. The code sample below already goes into manual code instrumentation.

I feel this is useful info, but reality is that the majority of the people will be happy with the code snippet above and won't do manual instrumentation.

Reply:
> We are in a Quick start section here, and honestly I'm not sure where to stop.

😂 I can relate to this.